### PR TITLE
Provide backend for seasonal statistics

### LINKED ISSUE #255: Add automatic seasons
The alternative to automatic seasons would be just deleting the database every time which would be bad in the sense that we would lose all data every few months. Instead with automatic seasons we can easily just back up everything in an alternate table and delete the old stats/matches afterwards. This allows us to preserve the data and would give us the option to present it somehow later.

NOTE: This issue has already been dealt with in #249 and it's only for documenting purposes.

### PR DESCRIPTION
**Pull request by czarnecki**
The old statistics are save in an extra table. Following that all
statistics are reset and all Matches are deleted. Any remaining
notifications should be deleted as well.
There is currently no option to query any of the seasons but that should
be easy to implement when necessary.
When the season has ended a "SeasonEndedEvent" is sent out. This should
allow us to for example send emails when a season has ended.
The current cron job has seasons set to 3 months, resetting on the first of each month at 01:00

Closes #255

### CODE REVIEW

File: src/main/java/de/adesso/kicker/season/persistence/SeasonStatistic.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+import javax.persistence.*;
+import java.util.List;
+
+@Builder
+@AllArgsConstructor
+@NoArgsConstructor
+@Data
+@Entity
+public class SeasonStatistic {
```
Comment (Flo4l) [inline src/main/java/de/adesso/kicker/season/persistence/SeasonStatistic.java]: Not sure if Both classes are neccessary. 
If we would only provide the `SeasonTrackedStatistic`s that would contain the purpose of this class, if a last object would be created right before resetting the season.

The other (not so nice) way would be to only save the 'SeasonStatistic' to provide only the really relevant information.

The only way both classes makes sense would be if we would provide the option to revisit profiles for older seasons.

File: src/main/java/de/adesso/kicker/season/service/SeasonService.java [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+    private void deleteOldSeason() {
+        matchService.deleteAll();
+        logger.info("All matches have been deleted");
+
+        userService.deleteAllStatistics();
+        logger.info("All statistics have been deleted");
+    }
+
+    private void seasonEndedEvent(String seasonName) {
```
Comment (Flo4l) [inline src/main/java/de/adesso/kicker/season/service/SeasonService.java]: I'd like to have this named `sendSeasonEndedEvent`

Comment (Flo4l) [reply src/main/java/de/adesso/kicker/season/service/SeasonService.java]: Also may i'm overlooking it but is there any listener for this event yet?

**Comment by Flo4l** [comment]
Good work!
I think the `SeasonEndedEvent` could be used to send a message to each user containing a message like
`Season x has ended. You Reached Rank y. Good luck and have fun in Season (x+1)`

File: src/main/java/de/adesso/kicker/season/service/SeasonService.java (hunk 2 above)
Comment (czarnecki) [reply src/main/java/de/adesso/kicker/season/service/SeasonService.java]: I didn't implement any listener's because that wan't of my concern.

**Pull request merged**